test/src/org/labkey/test/tests/targetedms/TargetedMSChromatogramOptimizationTest.java

```diff
@@ -0,0 +1,126 @@
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        TargetedMSChromatogramOptimizationTest init = (TargetedMSChromatogramOptimizationTest) getCurrentTest();
+        init.setupFolder(FolderType.Library);
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return "TargetedMS Chromatogram Optimization Test";
+    }
+
+    @Test
+    public void testUpload() throws Exception
+    {
+        goToProjectHome();
+        importData(SKY_FILE);
+        clickAndWait(Locator.linkContainingText("Panorama Dashboard"));
+        File downloadedClibFile = doAndWaitForDownload(() -> clickButton("Download", 0));
+
+        goToProjectHome();
+        goToSchemaBrowser();
+        DataRegionTable table = viewQueryData("targetedms", "TransitionOptimization");
+        table.rowSelector().showAll();
+
+        checker().verifyEquals("Invalid number of rows in transition optimization", table.getDataRowCount(),
+                sizeOfTable(downloadedClibFile, "MoleculeTransitionOptimization"));
+
+        log("Verifying the new tables added");
+        checker().verifyTrue("TransitionOptimization table is not present in the SQLITE file",
+                tableExists(downloadedClibFile, "TransitionOptimization"));
+        checker().verifyTrue("MoleculeTransitionOptimization table is not present in SQLITE file",
+                tableExists(downloadedClibFile, "MoleculeTransitionOptimization"));
+
+        log("Verifying the SampleFile modifications");
+        checker().verifyTrue("Sample File does not have CePredictorId",
+                columnExists(downloadedClibFile, "SampleFile", "CePredictorId"));
+        checker().verifyTrue("Sample File does not have DpPredictorId",
+                columnExists(downloadedClibFile, "SampleFile", "DpPredictorId"));
+
+    }
+
+    private int sizeOfTable(File clibFile, String name)
+
+    {
+        int cnt = 0;
+        @SuppressWarnings("SqlResolve")
+        String sql = "SELECT * FROM " + name;
+        try (
+                Connection conn = ConnectionSource.getConnection(clibFile.getAbsolutePath()))
+        {
+            ResultSet rs = conn.createStatement().executeQuery(sql);
+            while (rs.next())
+                cnt++;
+        }
+        catch (SQLException e)
+        {
+            throw new RuntimeException(e);
+        }
+        return cnt;
+    }
+
+    private boolean tableExists(File clibFile, String tableName)
+    {
+        try (
+                Connection conn = ConnectionSource.getConnection(clibFile.getAbsolutePath()))
+        {
+            DatabaseMetaData md = conn.getMetaData();
+            ResultSet rs = md.getTables(null, null, tableName, null);
+            if (rs.next())
+                return true; //Table exists
+            else
+                return false;
+        }
+        catch (SQLException e)
+        {
+            throw new RuntimeException(e);
```

Comment:
This is an improvement over just catching the exception and printing the stack trace, but you can declare that these methods throw SQLException and not have to catch and rethrow them. The method that's calling them is already throwing Exception, so this will reduce the code footprint.

Reply:
Done.